openedx/features/colaraz_features/helpers.py

```diff
@@ -0,0 +1,115 @@
+
+    Returns:
+        (Pair<Site, Site>): named tuple containing lms, studio sites.
+    """
+    sites = (
+        Site.objects.create(
+            domain='{}.{}'.format(prefix, domain),
+            name=name
+        ) for prefix in ('courses', 'studio')
+    )
+    return Pair(*sites)
+
+
+def create_site_themes(sites, theme):
+    """
+    Create site themes for the site provided in the arguments.
+
+    Arguments:
+        sites (Pair<Site, Site>): Named tuple containing lms and studio sites.
+        theme (str): Theme name to be added for the sites.
+    Returns:
+        (Pair<SiteTheme, SiteTheme>): named tuple containing lms, studio sites themes.
+    """
+    site_themes = (
+        SiteTheme.objects.create(
+            site=site,
+            theme_dir_name=theme
+        ) for site in sites
+    )
+    return Pair(*site_themes)
+
+
+def create_organization(name, short_name):
+    """
+    Create an organization with the given name.
+
+    Short name for the organization will be extracted from the name by slugify-ing the name.
+
+    Arguments:
+        name (str): Name of the organization to create.
+        short_name (str): Short name of the organization to create.
+
+    Returns:
+        (organization.Organization): Instance of the newly created organization.
+    """
+    return Organization.objects.create(
+        name=name,
+        short_name=short_name,
+    )
+
+
+def create_site_configurations(sites, organization, university_name, platform_name, organizations):
+    """
+    Create site configurations for the given sites.
+
+    Following fields will be set in the configuration
+        1. SESSION_COOKIE_DOMAIN: site.domain
+        2. PLATFORM_NAME: platform_name
+        3. site_domain: site.domain
+        4. site_name: university_name
+        5. course_org_filters: [ organization.short_name ] + organizations
+
+    Arguments:
+        sites (Pair<Site, Site>): named tuple containing lms and studio sites for which to create the configuration.
+        organization (organization.Organization): Organization to associate with the configuration.
+        university_name (str): Name of the university
+        platform_name (str): Platform name that appears on the dashboard and several other places.
+        organizations ([str]): List of organizations to add in the configuration.
+
+    Returns:
+        (Pair<SiteConfiguration, SiteConfiguration>): named tuple containing lms, studio sites configurations.
+    """
+    site_configurations = (
+        SiteConfiguration.objects.create(
+            site=site,
+            enabled=True,
+            values={
+                'SESSION_COOKIE_DOMAIN': site.domain,
+                'PLATFORM_NAME': platform_name,
+                'platform_name': platform_name,
+                'site_domain': site.domain,
+                'site_name': university_name,
+                'course_org_filters': organizations + [organization.short_name]
```

Comment:
Is there any scenario where we get duplicate values?

Reply:
Done